v2/internal/plugins/common/proxy/context.go

```diff
@@ -172,6 +172,13 @@ func (c *Context) ResetWarnings() {
 	c.sessionWarnings = []types.Warning{}
 }
 
+func (c *Context) RawContext() context.Context {
+	c.ctxMu.RLock()
```

Comment:
what's the point of locking since the internal ref is returned anyway?

Reply:
and what would happen when the c.ctx changes? the code that fetched the old one is gonna hang?

Reply:
task context is always canceled when collect is completed so fetch will not block forever. For a given context (related to task) context not changes during next Collect (it's just new context)

Regarding locking it's aligned to other context.Context related methods. Probably could return it without mock but wanted to leave it.